src/cloudant/account.py

```diff
@@ -392,32 +392,38 @@ def _usage_endpoint(self, endpoint, year=None, month=None):
         optional year and month URL elements.
 
         :param str endpoint: Cloudant usage endpoint.
-        :param int year: Year to query against.  Defaults to None.
-        :param int month: Month to query against.  Defaults to None.
-        """
-        if year is not None:
-            endpoint = posixpath.join(endpoint, str(year))
-        if month is not None:
-            if year is None:
-                raise CloudantException(
-                    (
-                        "must supply both year and month "
-                        "to usage endpoint: {0}"
-                    ).format(endpoint)
-                )
-            endpoint = posixpath.join(endpoint, str(month))
-        resp = self.r_session.get(endpoint)
+        :param int year: Year to query against.  Optional parameter.
+            Defaults to None.  If used, it must be accompanied by ``month``.
+        :param int month: Month to query against that must be an integer
+            between 1 and 12. Optional parameter. Defaults to None.
+            If used, it must be accompanied by ``year``.
+        """
+        if year and month in range(1, 13):
+            resp = self.r_session.get(
+                posixpath.join(endpoint, str(year), str(month))
+            )
+        elif not year and not month:
+            resp = self.r_session.get(endpoint)
+        else:
+            raise CloudantException(
+                (
+                    "must supply both year and a valid month "
+                    "between 1 and 12 to usage endpoint: {0}"
+                ).format(endpoint)
```

Comment:
I think this error message is misleading and the exception should be a `CloudantArgumentError` in this case.  I know it was from the original code but it doesn't seem to work here.  I would change this to:

```python
msg = 'Invalid year and/or month supplied.  Found: year - {0}, month - {1}'.format(year, month)
raise CloudantArgumentError(msg)
```